prisma/migrations/20240128111430_add_relation_as_db_field/migration.sql

```diff
@@ -0,0 +1,5 @@
+-- AlterTable
+ALTER TABLE "user_achievement" ADD COLUMN     "relation" TEXT;
+
+-- CreateIndex
+CREATE UNIQUE INDEX "user_achievement_relation_userId_templateId_key" ON "user_achievement"("relation", "userId", "templateId");
```

Comment:
What about achievements that can be achieved multiple times?

Reply:
Or can every achievement only be achieved once per relation?

Reply:
Yes. We can have multiple achievements of the same template, but they have to have different relations. Otherwise, we cannot distinguish them afterwards.